### DISCUSSION TITLE
Dorji DRF1268T Error -12

### DISCUSSION THREAD

**Discussion by ConnectedSensorsJohann** [jgromes/RadioLib#686] [Q&A]
Hey, this is not directly related I don't think to radiolib issue so I'll post it here instead of issue tracker to clarify 

I am trying to setup a TinyGS station with custom ESP32 and DRF1268T module. Managed to track down the required pin maps on the TinyGD build and can get the system up and running successfully. But only until I want to chnage freq or even the auto tuner wants to change frequency settings then system returns error -12. What I can find in wiki this is related to invalid frequency selection. 

What I don't understand is the same release works fine with other 1268 modules on bought boards ( Tbeam) and the new module is Tcxo enabled and has a frequencies opperation well beyond the 400 to 437mhz we need. So why is it returning -12 error. If I change the frequency to 433mhz its fine. Change it to 437 or 401 it gives same -12 error.

**Comment by jgromes** [comment]
The SX1268 frequency range is 410 - 810 MHz. This range check is performed by RadioLib, and that's what's returning the -12 error code. It is possible to suppress this check by a compile-time flag, however, the module itself will actually fail to calibrate if the frequency is out of bounds.

You should check that the module type is indeed SX1268, and if so, whether the frequency you are setting is actually within the range. Can you post a minimum working example that shows this issue?

**Comment by ConnectedSensorsJohann** [comment]
Thank you for the reply.

I could post the code here but like I mentioned it's the TinyGS stations so
there code base I just know they use radiolib. some satellites they are
tracking is 400mhz.

So if the module can't do it it can't do it. But what's confusing is some
tbeam and lilygo modules claim to be 1268 and they are able to work with
400. So if the flags are disabled will it just work as it's on the edge of
frequency range. Or is this a no go. Trying to understand if someone has
tested this

**Reply by jgromes** [reply]
I would suggest trusting the datasheet unless you have a proof that there is none or acceptable impact on operating outside of the specification range.

However there's a simpler solution - use the SX1262, as it covers 150 - 960 MHz.

**Comment by Rustie0125** [comment]
I totally. Agree I just keep speaking to suppliers of 1262 modules and they
all advised against running a module from 150 to 960mhz.

Obviously some hardware optimizations per set freq band but overall is this
a workable solution if I have a module and program it to any freq? Apart
from antenna change and optimization what else stops someone from building
or using a 1262 as a wideband module.

Context: I'm reviewing this for custom build tinygs station and weather
Balloon so trying to either find best compromise instead of adding three
radio modules to the same board.

**Reply by jgromes** [reply]
> Apart from antenna change and optimization what else stops someone from building
or using a 1262 as a wideband module.

You will need some impedance matching network between the SX1262 and the antenna. The impedance of this network is frequency-dependent, that's why you can't really use it as wideband transceiver. You could have multiple impedance networks + anteannas for different bands, but then you would also need some RF switch to change between them, which will have some losses.

In addition, the module itself also performs calibration on startup, for a specific frequency band. So even if you manage to resolve the issues above, you would have to recalibrate the module whenever you change the band if the difference is more than a couple MHz.

**Comment by ConnectedSensorsJohann** [comment]
Ps same person here.. Google just decided what account to use when it feels
like it 😉

**Comment by Rustie0125** [comment]
Yes I do get the part of the losses. And if these loses are 10 to 15% might
be worth it.

There are plenty of wideband baluns available that covers the 200 to 1ghz
bands which should do a decent job at impedance matching some below 1db
insertion loss.

The calibration part is new to me. Is this a question of reinitlizing the
radio after band change or is some external process/calibration required.

**Reply by jgromes** [reply]
You can do it by a command, so it can be done from software, however, the SX126x devices seems somewhat picky about the order of commands, it is possible you might have to basically perform initialization again. Also, you will not be able to receive or transmit in multiple bands simultaneously.

It's definitely not something I have ever tried.

**Comment by Rustie0125** [comment]
Okay.. intresting.

I'm my daily work we use sx1262 and completely shutdown and re inti with
every wakeup. Only way to get the device to sleep elbow 10uA we found but
that lorawan and don't use radio lib.

I don't see issue with recal and re initial when changing bands. I'm
thinking for a sub 10g Pico Balloon moving across different ISM bands and
auto switching this way is worth the weight, cost and size. Band switch
will basically be initiated by 3 or four programmed waypoints and 500km
radius around those points so should not happen to often.

**Comment by ConnectedSensorsJohann** [comment]
Would make for a intresting example....  calling radio function by ISM band
and having all this done automatically.

Lol pushing my luck here.

Thanx for the help. Always helps bouncing these ideas around.
